client/src/lib/api.js

```diff
@@ -0,0 +1,231 @@
+ * @returns Promise that resolves to an array of exercise logs.
+ */
+export async function fetchUserLogs(id) {
+  const response = await fetch(`/api/exercises/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches user's groups from the groups table in the API.
+ * @param {Number} 'id' - an integer representing the userId.
+ * @returns Promise that resolves to an array of groups.
+ */
+export async function fetchGroups(id) {
+  const response = await fetch(`/api/groups/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches group exercises from the API.
+ * @param {Number} 'id' - an integer representing the groupId.
+ * @returns Promise that resolves to an array of exercise logs for all group members.
+ */
+export async function fetchGroupLogs(id) {
+  const response = await fetch(`/api/group-logs/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches group data from the API.
+ * @param {Number} 'id' - an integer representing the groupId.
+ * @returns Promise that resolves to an array of group settings data.
+ */
+export async function fetchGroupSettings(id) {
+  const response = await fetch(`/api/group-settings/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches group data from the API.
+ * @param {Number} 'id' - an integer representing the groupId.
+ * @returns Promise that resolves to an array of group penalties data.
+ */
+export async function fetchGroupPenalties(id) {
+  const response = await fetch(`/api/group-penalties/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches penalties data from the API.
+ * @param {Number} 'id' - an integer representing the userId.
+ * @returns Promise that resolves to an array of the user's penalties.
+ */
+export async function fetchUserPenalties(id) {
+  const response = await fetch(`/api/user-penalties/${id}`);
+  if (!response.ok) {
+    throw new Error(`Status ${response.status}`);
+  }
+  return await response.json();
+}
+
+/**
+ * Fetches all login information and verifies account from the API.
+ * @returns Promise that resolves to an object with a token and user information.
+ */
+export async function postAccount(account) {
+  const response = await fetch('/api/sign-in', {
+    method: "POST",
+    headers: { "Content-Type": "application/json" },
+    body: JSON.stringify(account)
+  });
+  if (!response.ok) {
+    return;
+  }
+  return await response.json();
+}
+
+/**
+ * Posts new exercise entry into the exercises table in the API.
+ * @param {Object} 'newLog' - an object with data collected from the form.
+ */
+export async function postNewAccount(account) {
```

Comment:
On all of these `post` methods, I don't like the use of `post` in the function name. That says too much about the implementation. What if you changed your API and made them `put`s instead of `post`s? Or used a completely different technique? `registerAccount` would be a better name here, and `addExerciseLog` would be a better name below, and so forth.

Try to make your function names semantically meaningful. That is, they should describe _what_ they are doing -- not _how_ they are doing it.